api/grpcserver/mesh_service.go

```diff
@@ -0,0 +1,360 @@
+		meshTxIds := s.getTxIdsFromMesh(startLayer, addr)
+		mempoolTxIds := s.Mempool.GetTxIdsByAddress(addr)
+
+		// Look up full data for all unique txids
+		txs, missing := s.Mesh.GetTransactions(append(meshTxIds, mempoolTxIds...))
+
+		// TODO: Do we ever expect txs to be missing here?
+		// E.g., if this node has not synced/received them yet.
+		if len(missing) != 0 {
+			log.Error("could not find transactions %v", missing)
+			return nil, status.Errorf(codes.Internal, "error retrieving tx data")
+		}
+		for _, t := range txs {
+			res.Data = append(res.Data, &pb.AccountMeshData{
+				DataItem: &pb.AccountMeshData_Transaction{
+					Transaction: convertTransaction(t),
+				},
+			})
+		}
+	}
+
+	// Gather activation data
+	if in.Filter.AccountMeshDataFlags&uint32(pb.AccountMeshDataFlag_ACCOUNT_MESH_DATA_FLAG_ACTIVATIONS) != 0 {
+		// TODO: we have no way to look up activations by coinbase
+		// so we have no choice but to read all of them (!!!)
+		var activations []types.ATXID
+		for l := startLayer; l <= s.Mesh.LatestLayer(); l++ {
+			layer, err := s.Mesh.GetLayer(l)
+			if layer == nil || err != nil {
+				return nil, status.Errorf(codes.Internal, "error retrieving layer data")
+			}
+
+			for _, b := range layer.Blocks() {
+				activations = append(activations, b.ATXIDs...)
+			}
+		}
+
+		// Look up full data
+		atxs, matxs := s.Mesh.GetATXs(activations)
+		if len(matxs) != 0 {
+			log.Error("could not find activations %v", matxs)
+			return nil, status.Errorf(codes.Internal, "error retrieving activations data")
+		}
+		for _, atx := range atxs {
+			// Filter here, now that we have full data
+			if atx.Coinbase == addr {
+				pbatx, err := convertActivation(atx)
+				if err != nil {
+					return nil, status.Errorf(codes.Internal, "error serializing activation data")
+				}
+				res.Data = append(res.Data, &pb.AccountMeshData{
+					DataItem: &pb.AccountMeshData_Activation{
+						Activation: pbatx,
+					},
+				})
+			}
+		}
+	}
+
+	res.TotalResults = uint32(len(res.Data))
+	return res, nil
+}
+
+func (s MeshService) getTxIdsFromMesh(minLayer types.LayerID, addr types.Address) []types.TransactionID {
+	var txIDs []types.TransactionID
+	for layerID := minLayer; layerID < s.Mesh.LatestLayer(); layerID++ {
+		destTxIDs := s.Mesh.GetTransactionsByDestination(layerID, addr)
+		txIDs = append(txIDs, destTxIDs...)
+		originTxIds := s.Mesh.GetTransactionsByOrigin(layerID, addr)
+		txIDs = append(txIDs, originTxIds...)
+	}
+	return txIDs
+}
+
+func convertTransaction(t *types.Transaction) *pb.Transaction {
+	return &pb.Transaction{
+		Id: &pb.TransactionId{Id: t.ID().Bytes()},
+		Data: &pb.Transaction_CoinTransfer{
+			CoinTransfer: &pb.CoinTransferTransaction{
+				Receiver: &pb.AccountId{Address: t.Recipient.Bytes()},
+			},
+		},
+		Sender: &pb.AccountId{Address: t.Origin().Bytes()},
+		GasOffered: &pb.GasOffered{
+			// TODO: Check this math! Does fee == price?
+			GasPrice:    t.Fee,
```

Comment:
@antonlerner does `transaction.Fee` mean gas price? Or something else?

Reply:
we currently don't distinguish gas and gas price, we have one fee property for simple tx

Reply:
Okay. I will not set GasPrice here since we don't currently implement it. Fee will be used in the GlobalStateService and transaction receipts, which contain post-STF data.